buildpack_integration_test.go

```diff
@@ -0,0 +1,166 @@
+
+	. "github.com/onsi/gomega"
+
+	"github.com/paketo-buildpacks/occam"
+	. "github.com/paketo-buildpacks/occam/matchers"
+	"github.com/paketo-buildpacks/packit/v2/pexec"
+)
+
+const lifecycleVersion = "0.14.0"
+
+func testBuildpackIntegration(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		buildPlanBuildpack string
+		goDistBuildpack    string
+
+		builderConfigFilepath string
+
+		pack    occam.Pack
+		docker  occam.Docker
+		source  string
+		name    string
+		builder string
+
+		image     occam.Image
+		container occam.Container
+	)
+
+	it.Before(func() {
+		pack = occam.NewPack().WithVerbose()
+		docker = occam.NewDocker()
+
+		var err error
+		name, err = occam.RandomName()
+		Expect(err).NotTo(HaveOccurred())
+
+		buildpackStore := occam.NewBuildpackStore()
+
+		buildPlanBuildpack, err = buildpackStore.Get.
+			Execute("github.com/paketo-community/build-plan")
+		Expect(err).NotTo(HaveOccurred())
+
+		goDistBuildpack, err = buildpackStore.Get.
+			WithVersion("1.2.3").
+			Execute("github.com/paketo-buildpacks/go-dist")
+		Expect(err).NotTo(HaveOccurred())
+
+		source, err = occam.Source(filepath.Join("integration", "testdata", "simple_app"))
+		Expect(err).NotTo(HaveOccurred())
+
+		builderConfigFile, err := os.CreateTemp("", "builder.toml")
+		Expect(err).NotTo(HaveOccurred())
+		builderConfigFilepath = builderConfigFile.Name()
+
+		_, err = fmt.Fprintf(builderConfigFile, `
+[lifecycle]
+  version = "%s"
+
+[stack]
+  build-image = "%s:latest"
+  id = "io.buildpacks.stacks.jammy"
+  run-image = "%s:latest"
+`,
+			lifecycleVersion,
+			stack.BuildImageID,
+			stack.RunImageID,
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(docker.Pull.Execute(fmt.Sprintf("buildpacksio/lifecycle:%s", lifecycleVersion))).To(Succeed())
+
+		Expect(archiveToDaemon(stack.BuildArchive, stack.BuildImageID)).To(Succeed())
+		Expect(archiveToDaemon(stack.RunArchive, stack.RunImageID)).To(Succeed())
+
+		builder = fmt.Sprintf("builder-%s", uuid.NewString())
+		logs, err := createBuilder(builderConfigFilepath, builder)
+		Expect(err).NotTo(HaveOccurred(), logs)
+	})
+
+	it.After(func() {
+		Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
+		Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+		Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+
+		Expect(docker.Image.Remove.Execute(builder)).To(Succeed())
+		Expect(os.RemoveAll(builderConfigFilepath)).To(Succeed())
+
+		Expect(docker.Image.Remove.Execute(stack.BuildImageID)).To(Succeed())
+		Expect(docker.Image.Remove.Execute(stack.RunImageID)).To(Succeed())
+
+		Expect(os.RemoveAll(source)).To(Succeed())
+	})
+
+	it("builds an app with a buildpack", func() {
+
+		var err error
+		var logs fmt.Stringer
+		image, logs, err = pack.WithNoColor().Build.
+			WithPullPolicy("never").
+			WithBuildpacks(
+				goDistBuildpack,
+				buildPlanBuildpack,
+			).
```

Comment:
What was your rationale for testing this stack with the Go buildpack? To me, I like the approach that Victoria took on the Bionic Full stack PR by testing with PHP, because PHP needs the full stack, whereas this app could work on Tiny, Base or Full right? This accomplishes two things: it shows that the stack is properly formed and can be used in a build, and second it show that the stack aligns with the expectations of what a Full stack should be able to build.

Maybe that second point is more what the builder tests are for, though. Interested to hear your thoughts. I also realize that PHP on Jammy isn't supported yet, but maybe a Node app with a `node-gyp` module would work.

Reply:
Sorry, I think we discussed this in-person but didn't share my thinking in writing.

My plan was to get an initial jammy-full-stack built, then validate php against it, release a php buildpack with the jammy stack support, and then update these integration tests to use that. I just wanted to start with a simple app to validate that the stack builds something.

I'm not averse to `node-gyp`, I'm just less familiar with it. If you think it's easy to create an app that clearly uses those features then we can use it from the beginning instead of the two-step approach i outline above.

Reply:
To be explicit - I think the integration tests for this stack should demonstrate both things you mentioned:
1. The stack can be built and executed
2. The stack contains packages that do not exist in tiny/base

So I think a single test with one single app that requires packages not present in tiny/base is sufficient. I'm ok with either Php or `node-gyp` (or any other option we identify).

Reply:
@robdimsdale that plan seems completely reasonable. Fine with the PHP approach you outlined or the Node approach (I think [this app](https://github.com/paketo-buildpacks/npm-install/tree/b6d80246cfad797ea67c584a17b3d3b07eef6604/integration/testdata/vendored) is an example, as explained in [this section](https://github.com/paketo-buildpacks/npm-install/tree/b6d80246cfad797ea67c584a17b3d3b07eef6604#stack-support) of the README). As I said though, whichever way you want to do it works for me.